spec/syntax.md

```diff
@@ -393,30 +394,80 @@ operand    = literal / variable
 
 ### Annotation
 
-An **_annotation_** is part of an _expression_ containing either 
+An **_<dfn>annotation</dfn>_** is part of an _expression_ containing either 
 a _function_ together with its associated _options_, or
 a _private-use_ or _reserved_ sequence.
 
 ```abnf
 annotation = (function *(s option)) / reserved / private-use
 ```
 
-### Function
+#### Function
+
+A **_<dfn>function</dfn>_** is named functionality in an _annotation_.
+_Functions_ are used to evaluate, format, select, or otherwise process data
+values during formatting.
+
+Each _function_ is defined by the runtime's _function registry_. 
+A _function_'s entry in the _function registry_ will define 
+whether the _function_ is a _selector_ or formatter (or both),
+whether an _operand_ is required, 
+what form the values of an _operand_ can take,
+what _options_ and _option_ values are valid, 
+and what outputs might result. 
+See [function registry](./) for more information.
+
+A _function_ MAY accept an _operand_.
+No other positional arguments are possible.
```

Comment:
> If we say "accept", doesn't that introduce a potential dependency between the function syntax definition and the function registry?

Not really? It introduces a dependency between a given function's implementation and the appearance of an operand.

This text (the original and either your or my formulation) speaks to what a function can do (it is not required to accept an operand, even if one is provided). The second sentence is redundant, but was preserved from the existing document.

I think either version is subtly misleading, since the actual state of affairs is more like: a function can appear in an annotation, which may or may not follow an operand in an expression--and the operand serves as input to the function. In the syntax, `function` is not actually the functionality: the `function` is just the **name** of the functionality.

So maybe we should deal with this in `annotation`, for example:

> An _annotation_ can appear in an _expression_ by itself or following a single _operand_.
> When following an _operand_, the _operand_ serves as input to the _annotation_.

Then we should eliminate discussion here of operands.

Reply:
The annotation text sounds like a good approach; I'd much rather do that than argue about semantics.